### PR TITLE
17 user didnt activate send x days and delete

### PR DESCRIPTION
**Pull request by eyalgolan**
fixed:
* email sending bug - added authentication function in order to refresh tokens
added:
* sending email reminder for unactivated users, deleting unactivated users after X days

### CODE REVIEW

File: src/server/src/jobs/ActivationReminderJob.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+/*
+* delete all users that didnt activate after DAYS_DELETE_UNACTIVED_USERS
+*/
+const removeOldNotActivated = async () => {
+    let now = new Date(); // get current time
+
+    //set min time
+    let min = new Date(); 
+    const gracePeriod = 24*60*60 * 1000 * process.env.DAYS_DELETE_UNACTIVED_USERS
```
Comment (teddy4445) [inline src/server/src/jobs/ActivationReminderJob.js:27]: hold the calculated number and put in a comment the calculation process (the one now presented)

File: src/server/src/jobs/ActivationReminderJob.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    min.setTime(now.getTime() - gracePeriod);
+
+    // go over all unactivated users and delete all that didnt activate in the grace period
+    for await (const doc of User.find({creationDate: {$gt: min, $lt: now}})) {
```
Comment (teddy4445) [inline src/server/src/jobs/ActivationReminderJob.js:31]: Maybe gather a list of all the users we wish to delete and execute the delete command in the DB level one time?

File: src/server/src/jobs/ActivationReminderJob.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        { expiresIn: "60m" });
+    return token;
+};
+
+
+/*
+* cron job that runs every day and sends email activation reminder
+* to users that didnt activate their account
+*/
+export const scheduleJobs = async () => {
```
Comment (teddy4445) [inline src/server/src/jobs/ActivationReminderJob.js:71]: Maybe not each day but each REMIND_USERS_TO_ACTIVATE_DAYS

File: src/server/src/resources/emails.js [hunk 4]
```diff
@@ -39,3 +39,17 @@ export const generateForgotPasswordEmail = (email, name, token) => {
             The Lecture Website App Generator team</p>`,
   };
 };
+
+export const generateNewNotActivatedEmail = (email, name, token) => {
+  return {
+    from: "Lecture Website App Generator <eyalgolan96@gmail.com>",
+    to: email,
+    subject: `Hi ${name}, please verify your Lecture Website App Generator account`,
```
Comment (teddy4445) [inline src/server/src/resources/emails.js:47]: Get this text from Ayelet

**Review by teddy4445 — COMMENTED** [review]
I think this is fine - small improvements required before merging...

File: src/server/src/jobs/ActivationReminderJob.js (hunk 1 above)
Comment (eyalgolan) [reply src/server/src/jobs/ActivationReminderJob.js]: fixed

File: src/server/src/jobs/ActivationReminderJob.js (hunk 2 above)
Comment (eyalgolan) [reply src/server/src/jobs/ActivationReminderJob.js]: done

File: src/server/src/jobs/ActivationReminderJob.js (hunk 3 above)
Comment (eyalgolan) [reply src/server/src/jobs/ActivationReminderJob.js:69]: done

File: src/server/src/resources/emails.js (hunk 4 above)
Comment (eyalgolan) [reply src/server/src/resources/emails.js:47]: I've contacted her about this in the past. I'll check with her again.

**Pull request merged**